dandi/publish/tests/test_asset.py

```diff
@@ -73,3 +75,50 @@ def test_asset_rest_retrieve(api_client, asset):
         'updated': TIMESTAMP_RE,
         'metadata': asset.metadata,
     }
+
+
+@pytest.mark.django_db
+def test_asset_path(api_client, asset):
```

Comment:
Part of the purpose of moving this logic to the model layer is that the logic no longer lives in the API layer, so you don't need to involve the API to test it. You should be able to directly invoke `asset.get_path('/foo',['/foo/bar', '/bar'])` to test the functionality of the `get_path` function, and then just have a couple of token API tests to verify that the `@paths` action was implemented correctly.